client/js/models/archive.js

```diff
@@ -235,6 +237,24 @@ module.exports = {
       var archive = state.instance
       var readStream = archive.createFileReadStream(data.entryName)
       done(readStream)
+    },
+    download: function (data, state, send, done) {
+      const archive = state.instance
+      const zip = new Jszip()
+      zip.file('Hello.txt', 'Hello World\n')
+      Object.keys(state.entries).sort().forEach((key) => {
+        const entry = state.entries[key]
+        if (entry.type === 'directory') {
+          // XXX: might need to manually create things
```

Comment:
ideally these entries have some sort of reference to their full path in their object, if they don't already we should make that happen